### PR TITLE
Suppress clang error because of deprecation

### PR DESCRIPTION
**Pull request by ahcorde**

### CODE REVIEW

**Comment by ahcorde** [comment]
[Build Status](https://ci.ros2.org/view/nightly/job/nightly_linux_clang_libcxx/1960/)

File: test/test_filesystem_helper.cpp [hunk 1]
```diff
@@ -28,6 +28,10 @@ static constexpr const bool is_win32 = false;
 #if !defined(_WIN32)
 # pragma GCC diagnostic push
 # pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+# ifdef __clang__
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#endif
```
Comment (clalancette) [inline test/test_filesystem_helper.cpp:34]:
```suggestion
# endif
```

**Review by clalancette — CHANGES_REQUESTED** [review]
Minor thing, but otherwise looks good to me.

**Review by clalancette — APPROVED** [review]
Looks good to me with green CI on the regular CI, plus the Clang job.

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21551/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15836/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1046/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22267/)
* Clang [Build Status](http://ci.ros2.org/job/ci_linux_clang_libcxx/72/)

**Review by fujitatomoya — APPROVED** [review]

**Pull request merged**